crates/oxc_linter/src/rules/eslint/unicode_bom.rs

```diff
@@ -59,11 +59,15 @@ impl Rule for UnicodeBom {
         let has_bomb = source.starts_with('﻿');
 
         if has_bomb && matches!(self.bom_option, BomOptionType::Never) {
-            ctx.diagnostic(unexpected_unicode_bom_diagnostic(Span::new(0, 0)));
+            ctx.diagnostic_with_fix(unexpected_unicode_bom_diagnostic(Span::new(0, 0)), || {
+                return Fix::delete(Span::new(0, 3));
+            });
         }
 
         if !has_bomb && matches!(self.bom_option, BomOptionType::Always) {
-            ctx.diagnostic(expected_unicode_bom_diagnostic(Span::new(0, 0)));
+            ctx.diagnostic_with_fix(expected_unicode_bom_diagnostic(Span::new(0, 0)), || {
+                return Fix::new("﻿", Span::new(0, 0));
```

Comment:
Should there be a bom character in here, or is the github UI not showing?

Reply:
Yup, it completely hides it unfortunately

![image](https://github.com/oxc-project/oxc/assets/67428/a9018958-ff3e-43f9-9e04-142eddcb9a5d)